libraries/ESP_NOW/src/ESP32_NOW.cpp

```diff
@@ -0,0 +1,395 @@
+#include "ESP32_NOW.h"
+#include <string.h>
+#include "esp_system.h"
+#include "esp32-hal.h"
+#include "esp_mac.h"
+#include "esp_wifi.h"
+
+static void (*new_cb)(const esp_now_recv_info_t *info, const uint8_t * data, int len, void * arg) = NULL;
+static void * new_arg = NULL;// * tx_arg = NULL, * rx_arg = NULL,
+static bool _esp_now_has_begun = false;
+static ESP_NOW_Peer * _esp_now_peers[ESP_NOW_MAX_TOTAL_PEER_NUM];
+
+static esp_err_t _esp_now_add_peer(const uint8_t *mac_addr, uint8_t channel, wifi_interface_t iface, const uint8_t *lmk, ESP_NOW_Peer * _peer=NULL){
+    log_v(MACSTR, MAC2STR(mac_addr));
+    if (esp_now_is_peer_exist(mac_addr)) {
+        log_e("Peer Already Exists");
+        return ESP_ERR_ESPNOW_EXIST;
+    }
+
+    esp_now_peer_info_t peer;
+    memset(&peer, 0, sizeof(esp_now_peer_info_t));
+    memcpy(peer.peer_addr, mac_addr, ESP_NOW_ETH_ALEN);
+    peer.channel = channel;
+    peer.ifidx = iface;
+    peer.encrypt = lmk != NULL;
+    if(lmk){
+        memcpy(peer.lmk, lmk, ESP_NOW_KEY_LEN);
+    }
+
+    esp_err_t result = esp_now_add_peer(&peer);
+    if(result == ESP_OK){
+        if(_peer != NULL){
+            for(uint8_t i=0; i<ESP_NOW_MAX_TOTAL_PEER_NUM; i++){
+                if(_esp_now_peers[i] == NULL){
+                    _esp_now_peers[i] = _peer;
+                    return ESP_OK;
+                }
+            }
+            return ESP_FAIL;
+        }
+    } else if (result == ESP_ERR_ESPNOW_NOT_INIT) {
+        log_e("ESPNOW Not Init");
+    } else if (result == ESP_ERR_ESPNOW_ARG) {
+        log_e("Invalid Argument");
+    } else if (result == ESP_ERR_ESPNOW_FULL) {
+        log_e("Peer list full");
+    } else if (result == ESP_ERR_ESPNOW_NO_MEM) {
+        log_e("Out of memory");
+    } else if (result == ESP_ERR_ESPNOW_EXIST) {
+        log_e("Peer already exists");
+    }
+    return result;
+}
+
+static esp_err_t _esp_now_del_peer(const uint8_t *mac_addr){
+    esp_err_t result = esp_now_del_peer(mac_addr);
+    if (result == ESP_ERR_ESPNOW_NOT_INIT) {
+        log_e("ESPNOW Not Init");
+    } else if (result == ESP_ERR_ESPNOW_ARG) {
+        log_e("Invalid Argument");
+    } else if (result == ESP_ERR_ESPNOW_NOT_FOUND) {
+        log_e("Peer Not Found");
+    }
+
```

Comment:
shouldn't it return the error code as soon as it has failed?
or it can move on and execute the for() below?